src/components/modals/EntityCreatorEditor/EntityCreatorContainer.tsx

```diff
@@ -322,7 +322,7 @@ class Container extends React.Component<Props, ComponentState> {
             isMultivalueVal,
             isNegatableVal,
             entityTypeVal,
-            entityNameVal: isPrebuilt ? Container.GetPrebuiltEntityName(obj.text) : prevState.entityNameVal,
+            entityNameVal: isPrebuilt ? Container.GetPrebuiltEntityName(obj.text) : "",
```

Comment:
Seems like it would be a bit annoying to have is clear when it doesn't need to. I'd only clear it when switching away from a prebuilt

Reply:
Now that the new code detects the `builtin-` prefix I think when you switch away it will be an error anyways which would already force the user to change the name so maybe this isn't needed anymore. If we make it only clear on change from prebuilt it would make me wonder why it only clears the name some times.

Reply:
Blanket clearing feels is the lesser annoying behavior vs conditionally clearing the field based on the previous drop selection. The latter seems inconsistently quirky.